dev/com.ibm.ws.cdi.2.0.weld/src/com/ibm/ws/cdi/proxy/ProxyServicesImpl.java

```diff
@@ -40,149 +36,142 @@
 public class ProxyServicesImpl implements ProxyServices {
 
 private static final ManifestElement[] WELD_PACKAGES;
-	private static final ClassLoader CLASS_LOADER_FOR_SYSTEM_CLASSES = org.jboss.weld.bean.ManagedBean.class.getClassLoader(); //I'm using this classloader because we'll need the weld classes to proxy anything.
-
-	private static enum ClassLoaderMethods {
-		;//No enum instances
-
-		private static final Method defineClass1, defineClass2, getClassLoadingLock;
-
-		static {
-			try {
-				Method[] methods = AccessController.doPrivileged(new PrivilegedExceptionAction<Method[]>() {
-					public Method[] run() throws Exception {
```

Comment:
I think this part is more complicated.

We don't really want this static initializer to ever fail, regardless of the permissions of the code on the stack at the time of the call, since if it does fail then the class is completely unusable.

However, we do then need to make sure that no-one can abuse the `Method`s that we're pulling out, so we may need a check around where they're invoked.